scripts/issue/71/consolidate_badger.py

```diff
@@ -0,0 +1,145 @@
+
+        safe = GreatApeSafe(DEV_MULTISIG)
+        treasury_vault = GreatApeSafe(TREASURY_VAULT)
+        safe.take_snapshot(tokens=[registry.eth.treasury_tokens.BADGER])
+        treasury_vault.take_snapshot(tokens=[registry.eth.treasury_tokens.BADGER])
+
+        # Contracts needed
+        controller = interface.IController(
+            registry.eth.controllers.native,
+            owner=safe.account
+        )
+        balance_checker = interface.IBalanceChecker(
+            registry.eth.helpers.balance_checker,
+            owner=safe.account
+        )
+        dao_treasury = interface.ISimpleTimelockWithVoting(
+            registry.eth.badger_wallets.DAO_treasury,
+            owner=safe.account
+        )
+        bBadger_vault = interface.ISettV4h(
+            registry.eth.sett_vaults.bBADGER,
+            owner=safe.account
+        )
+        bBadger_strat = interface.IStrategy(
+            registry.eth.strategies["native.badger"],
+            owner=safe.account
+        )
+
+        ## == Release all BADGER from DAO_Treasury to devMulti == ##
+        
+        # Sets new beneficiary
+        dao_treasury.setBeneficiary(DEV_MULTISIG)
+        assert dao_treasury.beneficiary() == DEV_MULTISIG
+
+        treasury_balance = badger_token.balanceOf(dao_treasury.address)
+        gov_balance = badger_token.balanceOf(DEV_MULTISIG)
+
+        # Release BADGER to devMultisig
+        dao_treasury.release()
+
+        balance_checker.verifyBalance(
+            badger_token.address,
+            DEV_MULTISIG,
+            treasury_balance + gov_balance
+        )
+
+        ## == Transfer missing BADGER to Geyser and withdrawAll to vault == ##
+        geyser_balance = badger_token.balanceOf(BADGER_GEYSER)
+        strategy_balanceOf = bBadger_strat.balanceOf()
+        geyser_deficit = strategy_balanceOf - geyser_balance
+
+        # Ensure geyser is in deficit
+        assert geyser_deficit > 0
+        
+        # Transfer missing BADGER to Geyser
+        badger_token.transfer(BADGER_GEYSER, geyser_deficit, {"from": safe.address})
+
+        CONSOLE.print(f"{geyser_deficit / 1e18} BADGERs were transfer to the Geyser!\n")
+
+        balance_checker.verifyBalance(
+            badger_token.address,
+            BADGER_GEYSER,
+            geyser_deficit
+        )
+
+        # Once the Geyser has enough BADGER within, we can
+        # withdrawAll to transfer all BADGER from strat to the vault
+        assert bBadger_vault.available() == 0
+
+        controller.withdrawAll(bBadger_vault.token())
+
+        assert bBadger_vault.available() == strategy_balanceOf
+        assert bBadger_strat.balanceOf() == 0
+        assert badger_token.balanceOf(BADGER_GEYSER) == 0
+
+        # Transfer remaining of amount received from the DAO_treasury to
+        # the treaury vault multisig
+        amount = treasury_balance - geyser_deficit
+        badger_token.transfer(TREASURY_VAULT, amount, {"from": safe.address})
```

Comment:
this cannot be done precise onfork, i would opt for sending 8_500_000 and getting the remainder in another tx.

Reply:
2 txs approach sounds safest to avoid reverts from miscalc off-chain till signing/executing

Reply:
Very good call, I missed that.